homeassistant/components/vorwerk/config_flow.py

```diff
@@ -0,0 +1,128 @@
+"""Config flow to configure Vorwerk integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pybotvac.exceptions import NeatoException
+from requests.models import HTTPError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_CODE, CONF_EMAIL, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResultDict
+
+from . import api
+
+# pylint: disable=unused-import
+from .const import (
+    VORWERK_DOMAIN,
+    VORWERK_ROBOT_ENDPOINT,
+    VORWERK_ROBOT_NAME,
+    VORWERK_ROBOT_SECRET,
+    VORWERK_ROBOT_SERIAL,
+    VORWERK_ROBOT_TRAITS,
+    VORWERK_ROBOTS,
+)
+
+DOCS_URL = "https://www.home-assistant.io/integrations/vorwerk"
```

Comment:
This is already defined in the manifest file, so the proper way would be to access that information instead of hardcoding the url here again. Alas, I don't know how that's done, so I cannot help with it...

Reply:
> 
> 
> The device interactions (mostly everything in api.py, as well as the error code mappings in const.py) should be moved to the backend library.

Sorry, I must disagree here. The error mapping is just a mapping for the frontend and should be translated in the next step (next PR with the sensor and switch plattforms). IMHO translations and UI stuff should be done in home assistant and not in the underlying backend library.

The class `VorwerkState` is used to hold the state for the `UpdateDataCoordinator`. The Synology DSM, the PI-Hole integration and many more are using such classes. Often called API or Hub. It prepares the backend state for easier integration into the home assistant platforms. It is completely related to home assistant and should IMHO not be part of the underlying library.
To avoid confusion I'll remove the `api.py` module and move the classes according to the other integrations like Synology DSM and the PI-Hole .

Reply:
Having a class that sits on top of the third party library is quite normal and no problem. I think your idea to remove api.py and make something like a hub class is excellent and will save you a lot of questions.

Reply:
> 
> 
> This is already defined in the manifest file, so the proper way would be to access that information instead of hardcoding the url here again. Alas, I don't know how that's done, so I cannot help with it...

As the documentation is not written yet and most integrations don't have a link to the documentation, I will remove this.

Reply:
you well not get this merged without a PR with documentation in homeassistant.io